sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveQuerySuite.scala

```diff
@@ -1224,6 +1226,34 @@ class HiveQuerySuite extends HiveComparisonTest with SQLTestUtils with BeforeAnd
       }
     }
   }
+
+  test("SPARK-29037: Spark gives duplicate result when an application was killed") {
```

Comment:
Thanks for adding this. This looks like a good IT. If possible, 
- Can we have a concise UT at `core` module because you are touching `HadoopMapReduceCommitProtocol`?
- Otherwise, can we move this to `sql/core` module instead of `sql/hive` because you are using `using parquet`?